hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java

```diff
@@ -688,6 +688,29 @@ public <T extends Enum<T>> T getEnum(String name, T defaultValue) {
         rawConfig.getEnum(name, defaultValue));
   }
 
+  /**
+   * Returns the account-specific enum value if it exists, then
+   * looks for an account-agnostic value in case-insensitive manner.
+   * @param name Account-agnostic configuration key
+   * @param defaultValue Value returned if none is configured
+   * @param <T> Enum type
+   * @return enum value if one exists, else null
+   */
+  public <T extends Enum<T>> T getCaseInsensitiveEnum(String name, T defaultValue) {
+    String configValue = getString(name, null);
+    if (configValue != null) {
+      String normalizedConfigValue = configValue.toUpperCase();
+      for (T enumConstant : defaultValue.getDeclaringClass().getEnumConstants()) { // Step 3: Iterate over enum constants
+        if (enumConstant.name().equals(normalizedConfigValue)) {
```

Comment:
We are expecting that these would be in uppercase. What if we do equalsIgnoreCase()?

Reply:
Ohh yes, would be much better.
Thanks
Taken